src/preprocessing/preprocessing.py

```diff
@@ -29,33 +29,56 @@
 
 
 class Preprocessing:
-    def __init__(self, filter_null_data=True, historical_data=False):
+    def __init__(self, filter_null_data=True, historical_bool=True, S3_bool=False):

```

Comment:
Distinction between S3 or local shouldn't be done here. It should only depend on the Flag in the .env file. This leads to an error if DB type is local in the .env and S3_bool is True